### PR TITLE
feat(sdk): option to hide dashboard card title

### LINKED ISSUE #43819: Option to hide dashboard card title in embedding SDK
Expose the `withCardTitle` prop to allow hiding dashboard card titles. This should be `true` by default.

```[tasklist]
### Blockers
- [ ] https://github.com/metabase/metabase/issues/43973
```

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/43819

### Description

Adds the `withCardTitle` option to the `StaticDashboard` component which allows SDK users to hide the card title via `withCardTitle={false}`

### How to verify

- Navigate to the `test-card-title` branch in the demo app, and run the demo app
- Ensure that `routes/analytics/index.tsx` in the demo app has a `withCardTitle={false}` prop set in StaticDashboard
- Click on the Analytics Page. The card title should be hidden.

### Demo

CleanShot 2567-06-10 at 23 24 18@2x

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/public/StaticDashboard/StaticDashboard.unit.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  );
+};
+
+describe("StaticDashboard", () => {
+  it("should render", async () => {
+    setup();
+    expect(await screen.findByTestId("dashboard-grid")).toBeInTheDocument();
+    await waitForLoaderToBeRemoved();
+
+    // TODO: make dashcards render
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/StaticDashboard/StaticDashboard.unit.spec.tsx:106]: Will get help to fix this tomorrow. Multiple issues with getting this to render:

- The width controlled by ExplicitSize is always 0. We also pass in `width={0}` at the DashboardGridConnected call site.
- After passing a fixed width for testing, the loader never stops spinning.

**Review by deniskaber — APPROVED** [review]
Looks good!

File: frontend/src/metabase/dashboard/components/DashboardGrid.tsx [hunk 2]
```diff
@@ -133,6 +133,7 @@ type OwnProps = {
   isXray?: boolean;
   isFullscreen: boolean;
   isNightMode: boolean;
+  withCardTitle?: boolean;
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/dashboard/components/DashboardGrid.tsx:136]: Good catch!

File: enterprise/frontend/src/embedding-sdk/components/public/StaticDashboard/StaticDashboard.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+describe("StaticDashboard", () => {
+  it("shows a dashboard card question title by default", async () => {
+    setup();
+
+    await delay(1);
+
+    expect(await screen.findByTestId("dashboard-grid")).toBeInTheDocument();
+    expect(await screen.findByText("Here is a card title")).toBeInTheDocument();
+  });
```
Comment (kulyk) [inline enterprise/frontend/src/embedding-sdk/components/public/StaticDashboard/StaticDashboard.unit.spec.tsx:109]: Why do we need a delay here?

**Review by kulyk — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/components/public/StaticDashboard/StaticDashboard.unit.spec.tsx (hunk 3 above)
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/public/StaticDashboard/StaticDashboard.unit.spec.tsx:109]: @kulyk whoops, good catch! I was debugging something and forgot to remove it 🙏🏻

**Pull request merged**